internal/unionstore/art/art.go

```diff
@@ -300,11 +306,7 @@ func (t *ART) expandLeaf(key artKey, depth uint32, prev, current artNode) (arena
 	newAn.addChild(&t.allocator, l2Key.charAt(int(depth)), !l2Key.valid(int(depth)), leaf2Addr)
 
 	// swap the old leaf with the new node4.
-	if prev == nullArtNode {
```

Comment:
Why can we remove the check?

Reply:
Because the root node can't be a leaf, so when we expanding a leaf, the prev node is definitely not null.

Consider this case:
```
1. t.root == nullArtNode
2. insert a key [1, 2], t.root becomes a node4
3. insert a key [1, 3], t.root.findChild(1) is a leaf, so expandLeaf will be invoked, prev is t.root now
```